### PR TITLE
Supporting Jakarta Faces 3.0 in Config Files

### PR DESCRIPTION
**Pull request by gerdogdu**

### CODE REVIEW

**Comment by hussainnm** [comment]
The xsd files checked-in contain xi:include tags which are part of the source xsds. These xsds will not load properly because the .inc files are missing. Please use the published xsd available at https://jakarta.ee/xml/ns/jakartaee/

**Comment by gerdogdu** [comment]
Hi @hussainnm 
I have updated the XSD files.

File: impl/src/main/java/com/sun/faces/config/manager/DbfFactory.java [hunk 1]
```diff
@@ -256,49 +431,57 @@ public FacesEntityResolver() {
...
             // the systemId.
 
             // If no system ID, defer to superclass
             if (systemId == null) {
                 InputSource result;
                 try {
                     result = super.resolveEntity(publicId, null);
-                } catch (IOException | SAXException e) {
+                }
+                catch (IOException | SAXException e) {
```
Comment (arjantijms) [inline impl/src/main/java/com/sun/faces/config/manager/DbfFactory.java:462]: As much as I initially disliked the catch being on the same line of the closing brace, this is actually according to the Sun standard and the Jakarta EE standard (which inherits from Sun)

File: impl/src/main/java/com/sun/faces/config/manager/tasks/ParseConfigResourceToDOMTask.java [hunk 2]
```diff
@@ -109,10 +110,14 @@
      * Constructs a new ParseTask instance
      * </p>
      *
-     * @param servletContext the servlet context.
-     * @param validating whether or not we're validating
-     * @param documentURI a URL to the configuration resource to be parsed
-     * @throws Exception general error
+     * @param servletContext
```
Comment (arjantijms) [inline impl/src/main/java/com/sun/faces/config/manager/tasks/ParseConfigResourceToDOMTask.java:113]: I know it's a tiny nit and the rest of the Mojarra code is certainly not compliant, but the Jakarta EE coding standards ask to not indent javadoc. Its same line, or when too large (more than 120 chars) beginning of next line.

File: impl/src/main/java/com/sun/faces/config/manager/tasks/ParseConfigResourceToDOMTask.java [hunk 3]
```diff
@@ -78,6 +77,7 @@
 
     private static final String JAVAEE_SCHEMA_LEGACY_DEFAULT_NS = "http://java.sun.com/xml/ns/javaee";
     private static final String JAVAEE_SCHEMA_DEFAULT_NS = "http://xmlns.jcp.org/xml/ns/javaee";
+    private static final String JAKARTAEE_SCHEMA_DEFAULT_NS="https://jakarta.ee/xml/ns/jakartaee";
```
Comment (arjantijms) [inline impl/src/main/java/com/sun/faces/config/manager/tasks/ParseConfigResourceToDOMTask.java:80]: Almost too tiny to remark, but the coding standard asks for a space left and right of the assignment,

File: impl/src/main/java/com/sun/faces/config/manager/tasks/ParseConfigResourceToDOMTask.java [hunk 4]
```diff
@@ -247,9 +258,38 @@ private Document getDocument() throws Exception {
...
+                    default:
+                        throw new ConfigurationException("Unknown Schema version: " + versionStr);
+                    }
+                    DocumentBuilder builder = getBuilderForSchema(schema);
+                    if (builder.isValidating()) {
+                        builder.getSchema().newValidator().validate(domSource);
+                        returnDoc = ((Document) domSource.getNode());
                     } else {
-                        returnDoc = (Document) domSource.getNode();
+                        returnDoc = ((Document) domSource.getNode());
```
Comment (arjantijms) [inline impl/src/main/java/com/sun/faces/config/manager/tasks/ParseConfigResourceToDOMTask.java:292]: The extra braces are not needed here, just `(Document) domSource.getNode()` is enough

File: impl/src/main/java/com/sun/faces/config/manager/tasks/ParseConfigResourceToDOMTask.java [hunk 5]
```diff
@@ -347,16 +387,16 @@ private Document getDocument() throws Exception {
...
 
     private boolean streamIsZeroLengthOrEmpty(InputStream is) throws IOException {
-        boolean isZeroLengthOrEmpty = 0 == is.available();
+        boolean isZeroLengthOrEmpty = (0 == is.available());
         final int size = 1024;
         byte[] b = new byte[size];
         String s;
         while (!isZeroLengthOrEmpty && -1 != is.read(b, 0, size)) {
-            s = new String(b, RIConstants.CHAR_ENCODING).trim();
+            s = (new String(b, RIConstants.CHAR_ENCODING)).trim();
```
Comment (arjantijms) [inline impl/src/main/java/com/sun/faces/config/manager/tasks/ParseConfigResourceToDOMTask.java:395]: No need for the extra braces here.

**Comment by arjantijms** [comment]
Thanks a lot @gerdogdu! I provided some feedback on the formatting only, but they are very small nits and don't stand in the way of merging this. Thanks again! :)

**Comment by gerdogdu** [comment]
Your welcome @arjantijms . Actually I have been using Jakarta EE based Eclipse formatter.

**Comment by arjantijms** [comment]
@gerdogdu thanks again, no worries about the formatting ;)

File: impl/src/main/java/com/sun/faces/config/manager/tasks/ParseConfigResourceToDOMTask.java [hunk 6]
```diff
@@ -412,7 +455,7 @@ private DocumentBuilder getNonValidatingBuilder() throws Exception {
     }
 
     private DocumentBuilder getBuilderForSchema(Schema schema) throws Exception {
-        factory = DbfFactory.getFactory();
+        this.factory = DbfFactory.getFactory();
```
Comment (pzygielo) [inline impl/src/main/java/com/sun/faces/config/manager/tasks/ParseConfigResourceToDOMTask.java:458]: Why this `this.` was added here and not in other places?

**Pull request merged**